### PR TITLE
chore(ping): remove deprecated items

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

Related https://github.com/libp2p/rust-libp2p/issues/3647.

## Notes & open questions

Should be merged with next braking change to libp2p-ping.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by thomaseizinger** [comment]
We've now opened the merge window for breaking changes so this is good to go in once the merge conflicts are resolved.

File: protocols/ping/CHANGELOG.md [hunk 1]
```diff
@@ -2,8 +2,12 @@
 
 - Raise MSRV to 1.65.
   See [PR 3715].
+- Remove deprecated items. See [PR 3702].
+  * `Ping` prefixed items.
+  * Keep Alive option
```
Comment (thomaseizinger) [inline protocols/ping/CHANGELOG.md]:
```suggestion
```
 I'd like to keep this brief for less important changes.

File: protocols/ping/src/handler.rs [hunk 2]
```diff
@@ -103,25 +103,6 @@ impl Config {
...
-    ///
-    /// If the maximum number of allowed ping failures is reached, the
-    /// connection is always terminated as a result of [`ConnectionHandler::poll`]
-    /// returning an error, regardless of the keep-alive setting.
-    #[deprecated(
-        since = "0.40.0",
-        note = "Use `libp2p::swarm::behaviour::KeepAlive` if you need to keep connections alive unconditionally."
-    )]
-    pub fn with_keep_alive(mut self, b: bool) -> Self {
-        self.keep_alive = b;
```
Comment (thomaseizinger) [inline protocols/ping/src/handler.rs:121]: What about the field? Can we remove that too?

**Review by thomaseizinger — COMMENTED** [review]
Thanks!

File: protocols/ping/src/handler.rs (hunk 2 above)
Comment (umgefahren) [reply protocols/ping/src/handler.rs:121]: It get's read in `ConnectionHandler::connection_keep_alive`. We could probably just remove it though and always return that it shouldn't be kept alive. As one can't set it anymore anyway.

**Comment by thomaseizinger** [comment]
Can you fix the rustdoc issues? Thanks!

**Review by thomaseizinger — DISMISSED** [review]

**Review by mxinden — DISMISSED** [review]

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**